text/1001-websocket-protocol.rst

```diff
@@ -0,0 +1,139 @@
+..
+    Status: Draft
+    Type: Feature
+    Created: 2020-02-19
+    RFC PR: `edgedb/rfcs#4 <https://github.com/edgedb/rfcs/pull/4>`_
+
+======================================
+RFC 0001: The WebSocket-based Protocol
+======================================
+
+This RFC describes the design and implementation of the alternate
+protocol to the EdgeDB based on WebSockets for framing and the initial
+handshake.
+
+
+Motivation
+==========
+
+The primary motivation is to allow various browser-based exploration
+and editing tools for the database.
+
+The secondary motivation is to allow browser-based access to the
+application data. Essentially allowing serverless implementation for
+some applications or parts thereof. The reason this is a secondary
+motivation is that it requires full featured access controls (ACLs)
+which we don't have yet.
+
+Two important side effects of implementing this RFC:
+* it might also help implementing the protocol in some languages which
+  we don't have language bindings yet
+* existing tools like load-balancers can be used to access edgedb in
+  some cases
+* it enables using TLS which we don't have for the main protocol yet
```

Comment:
The main protocol is easy to extend with TLS as Postgres protocol demonstrates, so IMO this point is rather weak.

Reply:
Sure it's weak, that's why it's listed under "side effects" title. I don't think it makes sense to move it into another section, though.